src/main/java/org/testeditor/fixture/host/net/Connection.java

```diff
@@ -0,0 +1,227 @@
+            s3270Process.waitFor(5, TimeUnit.SECONDS);
+        } catch (final InterruptedException ex) {
+            logger.error("Something went wrong during termination of s2370 process");
+        }
+        try {
+            in.close();
+        } catch (final IOException ex) {
+            logger.error("Something went wrong during closing InputStreamreader of s2370 process");
+        }
+        cleanup();
+        logger.info("Disconnected successfully from host : {} ", hostname);
+        return !isConnected();
+    }
+
+    /**
+     * reset all opened streams
+     */
+    private void cleanup() {
+        out.close();
+        in = null;
+        out = null;
+        s3270Process = null;
+    }
+
+    /**
+     * perform a quit statement
+     */
+    private void doQuit() {
+        out.println("quit");
+        out.flush();
+    }
+
+    /**
+     * stop the s3270 process
+     */
+    private void stop2370Process() {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    Thread.sleep(1000);
+                } catch (final InterruptedException ex) {
+                    if (s3270Process != null) {
+                        s3270Process.destroy();
+                    }
+                    if (s3270Process != null) {
+                        s3270Process.destroy();
+                    }
+                }
+            }
+        }).start();
+    }
+
+    /**
+     *
+     * @return true if connection is established to mainframe through
+     *         s3270/ws3270 emulation and process is still running, false
+     *         otherwise.
+     */
+    public boolean isConnected() {
+        if (s3270Process == null || in == null || out == null) {
+            return false;
+        } else {
+            final Result r = doCommand("");
+            if (r.getStatus().matches(". . . C.*")) {
+                return true;
+            } else {
+                out.println("quit");
+                out.flush();
+                s3270Process.destroy();
+                s3270Process = null;
+                in = null;
+                out = null;
+                return false;
+            }
+        }
+    }
+
+    /**
+     * A s3270 command will be executed with this method. The whole
+     * communication with s3270 will be accessed through this method.
+     */
+    public Result doCommand(final String command) {
+        processAvailable();
+        try {
+            out.println(command);
+            out.flush();
+            logger.debug("*****************************************************************************************");
+            logger.debug("---> Command sent: \"{}\"", command);
+            List<String> lines = new ArrayList<String>();
+            readOutput(lines);
+            int size = lines.size();
+            if (size > 0) {
+                Result result = new Result(lines.subList(0, size - 1), lines.get(size - 1));
```

Comment:
why drop the list line? maybe put it into a separate method that describes why this is done?